pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -0,0 +1,140 @@
+package configuration
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/docker/distribution/configuration"
+)
+
+var (
+	// CurrentVersion is the most recent Version that can be parsed.
+	CurrentVersion = configuration.MajorMinorVersion(1, 0)
+
+	ErrUnsupportedVersion = errors.New("Unsupported openshift configuration version")
+)
+
+type openshiftConfig struct {
+	Openshift Configuration
```

Comment:
OpenShift but it really does sounds odd to have openshiftConfig.Openshift... do we really need that second openshift? Why not just call it conf?